include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -303,6 +304,10 @@ class fm_index
                                    | views::join(delimiter)
                                    | views::to<std::vector<uint8_t>>;
 
+        // we need at least one delimiter
+        if (std::ranges::size(text) == 1)
```

Comment:
Ok, I did quite a bit of debugging, because I wasn't entirely sure what line it originates from:

https://github.com/seqan/seqan3/blob/master/include/seqan3/search/fm_index/fm_index_cursor.hpp#L165

In the cursor we do `sigma(_index.index.sigma - index_t::text_layout_mode)`.
If we construct the sdsl index without delimiter, the sdsl's sigma is one smaller than it ought to be.
So we cancel the search too early because we think we iterated over the alphabet when extending.

I think it would be the easiest to just append the delimiter?

Reply:
`_index.index.sigma` 😭

Reply:
Ok, I slowly get to the point of it. Can you remove the special handling of a single element text and then always add the delimiter. That should not hurt in the performance I guess.